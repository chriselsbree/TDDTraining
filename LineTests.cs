using System.Linq.Expressions;
using ApprovalTests;

namespace Triangles
{
    [TestClass]
    public class LineTests
    {
        [TestMethod]
        public void TestVertical()
        {
            
            var line = new Line((1, 1), (1, 5));
            Assert.AreEqual(4,line.Length);

        }

        [TestMethod]
        public void TestHorizontal()
        {
            
            var line = new Line((5, 1), (8, 1));
            Assert.AreEqual(3, line.Length);

        }

        [TestMethod]
        public void TestDiagonal()
        {
            var line = new Line((0, 4), (3, 0));
            Assert.AreEqual(5, line.Length);
        }

        [TestMethod]
        public void TestLineWithEndpoints()
        {
            //Create a line with endpoints (1,5) and (6,5)
            //Verify that endpoints are (1,5) and (6,5)

            var line = new Line((1,5), (6,5));
            Approvals.VerifyAll("Endpoints", line.Endpoints,"Coordinate");

        }

        
       
    }
}