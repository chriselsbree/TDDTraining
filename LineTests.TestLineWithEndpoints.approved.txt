Endpoints

Coordinate[0] = (1,5)
Coordinate[1] = (6,5)
